### PR TITLE
Add `Cache` class

### PR DESCRIPTION
**Pull request by daniel-zint**
Implementing a `Cache` that automatically creates a unique directory where all temporary files can be written to. The folder is deleted automatically upon closure.

### CODE REVIEW

**Comment by mtao** [comment]
Just from looking over for paths, I'd put the cache stuff in the io folder

**Review by mtao — APPROVED** [review]
forgot to post these changes earlier - mostly good i think, there are minor points that dont matter too much i think

File: src/wmtk/utils/Cache.cpp [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    for (size_t i = 0; i < max_tries; ++i) {
+        const fs::path p =
+            m_cache_dir / (filename + "_" + timestamp + "_" + number_to_hex(i) + extension);
+
+        if (fs::exists(p)) {
+            continue;
+        }
+
+        // try to touch the file
+        std::ofstream ofs(p);
```
[the remark this answers is not in the document]
Comment (daniel-zint) [reply src/wmtk/utils/Cache.cpp:87]: I am using `std::ifstream` in `import_cache` so I need to include fstream anyways. I'll keep it the way it is. It is the nicer code and it should not have any impact on performance.

**Review by mtao — APPROVED** [review]
Lgtm

**Pull request merged**